contrib/gin-gonic/gin/option.go

```diff
@@ -46,3 +49,21 @@ func WithAnalyticsRate(rate float64) Option {
 		}
 	}
 }
+
+// WithResourceNamer specifies a function which will be used to obtain a resource name for a given
+// gin request, using the request's context.
+func WithResourceNamer(namer func(c *gin.Context) string) Option {
+	return func(cfg *config) {
+		cfg.resourceNamer = namer
+	}
+}
+
+// URLPathResourceNamer can be used for WithResourceNamer and returns the
+// request URL path as the resource name
+func URLPathResourceNamer(c *gin.Context) string {
```

Comment:
I like this, but it'd have very high cardinality (unique values).
So for now it's not something we can put in this contrib.
You'd still be able to write this in your codebase and set it as the resource namer if you chose to.

A suitable resource namer for this contrib with low cardinality would probably return the method and route, instead of the full path.
Eg: for `router.GET("/user/:id", ...)` when queried as `/user/123`, it'd return `GET /user/:id` or `GET /user/`.
I had a quick look and couldn't find a way to get the matched route from a `gin.Context`.

Reply:
I didn't think of the path vs matched route. It looks like this has been discussed in a gin isssue: https://github.com/gin-gonic/gin/issues/748 but there doesn't seem to be any movement on it. Some other authors have suggested workarounds to extract the matched route. Do you think a solution like that could be acceptable in contrib?

Reply:
Nice find! That issue is the exact problem and for very similar reasons.

The workarounds seem to have limitations, including the final one that attempts to rebuild the route based on the URL and Params. It'd work for some common cases, but isn't 100% accurate, eg https://play.golang.org/p/KYepJylHdtO

It'd be nicer to have the matched route value directly available in the `gin.Context`, since it would be 100% accurate.

Reply:
So leave it to the user?

Reply:
For now, yes.